Feature / 공통-컴포넌트-인풋(CheckBox, DatePicker, Avatar)

mieumje opened the pull request:
<!-- 
  PR 작성 시 해당 Issue 연결
  PR 이름: Feature/개발-완료-기능
  PR은 선착순 2명이 달면 수정해서 merge 
-->

## ⚙기능
<!-- 기능을 설명해주세요 -->
- 공통 컴포넌트 인풋 컴포넌트(CheckBox, DatePicker) 스토리북 내용 추가 148cae46db5176bdf682087b4fea5f8cfaeee17e
- 공통 컴포넌트 아바타 컴포넌트 스토리북 파일 추가 c980b64e00a601335fe490f9339af599e8ed8d3e dbc8870f3c707acf00b2d437f7736d3eec047c71
## 📃구현 내용
<!-- 구현 내용을 작성해 주세요 -->
- MuiInput 스토리북 파일에 CheckBox와 DatePicker 추가
- MuiAvatar 스토리북 파일 생성
## 💡기타
<!-- 추가로 적고 싶은 내용을 모두 적어주세요 -->
잘못된 내용 알려주시면 감사합니다!

- DatePicker
DatePicker의 경우 시작, 종료일을 한번에 지정하는 컴포넌트는 `@mui/x-date-pickers-pro`버전을 사용해야 한다고 해서 임의로 만들어 두었습니다. -> `MuiInput.stories.js -> StartAndEndDatePicker function`
- DatePicker Date Format
LA님과 협의 후 우선 date format을 'YYYY-MM-DD hh:mm:ss' 형태로 보낼 것 같습니다. 이 부분은 추후 변동이 있을 수 있습니다.
DatePicker의 선택된 날짜는 moment 객체로 moment js로 커스텀이 가능합니다.

mieumje committed:
chore: mui/lab 설치

datepicker 컴포넌트를 위해 mui/lab 패키지 설치

lexie-kaia committed:
style: MbtiTag 수정

lexie-kaia committed:
refactor: Mui 컴포넌트 스토리북 import 경로 수정

mieumje committed:
chore: mui/lab uninstall

mui/lab 내용이 mui/x-date-pickers로 이동되어 필요 없는 패키지 삭제

lexie-kaia committed:
refactor: 리뷰 반영

mieumje committed:
feat: CheckBox, DatePicker storybook 추가

mieumje committed:
feat: Avatar 컴포넌트 storybook 추가

mieumje committed:
refactor: Avatar 컴포넌트 수정

Text Avatar Group alt 지정
MuiProvider defaultProps width, height 삭제

lexie-kaia commented on src/components/Mui/MuiAvatar.stories.js:
```diff
@@ -0,0 +1,50 @@
+  variant: 'circular',
+  src: 'https://picsum.photos/id/1027/200/300',
+};
+
```
```
Default.argTypes = {
  variant: {
    options: ['circular', 'rounded', 'square', 'string'],
    control: { type: 'radio' },
  },
  src: {
    control: { type: 'text' },
  },
};
```
위에 export default 부분이 아니라
Default.argTypes에 추가하면
AvatarGroup에 불필요한 컨트롤을 만들지 않을 수 있습니다.

lexie-kaia replied:
그리고 variant를 모두 테스트하고 MuiProvider에서 전체 스타일을 입힌 것은 좋지만

아래에서 말한 것과 같은 이유로
저희 스토리북에 필요한 컨트롤 옵션인지 모르겠습니다

저는 Mui 컴포넌트 스토리북을 작성할 때
variant가 더 있어도 
다른 팀원에게 혼란을 줄 수 있어서
필요한 옵션이 아니면 넣지 않았습니다

mieumje replied:
> ```
> Default.argTypes = {
>   variant: {
>     options: ['circular', 'rounded', 'square', 'string'],
>     control: { type: 'radio' },
>   },
>   src: {
>     control: { type: 'text' },
>   },
> };
> ```
> 
> 위에 export default 부분이 아니라 Default.argTypes에 추가하면 AvatarGroup에 불필요한 컨트롤을 만들지 않을 수 있습니다.

새로 하나 배워갑니다 감사합니다!

> 저는 Mui 컴포넌트 스토리북을 작성할 때
variant가 더 있어도
다른 팀원에게 혼란을 줄 수 있어서
필요한 옵션이 아니면 넣지 않았습니다

이런 부분은 미처 생각하지 못했습니다.. 좋은 지적 감사합니다. 위 내용 참고해서 수정하겠습니다.

lexie-kaia commented on src/components/Mui/MuiInput.stories.js:
```diff
@@ -162,4 +168,180 @@ export const File = () => (
 );
 
-export const DatePicker = () => <MuiProvider></MuiProvider>;
+export const BasicCheckBoxes = () => {
```
mui checkbox의 사용법을 확인해본 것은 좋지만
스토리북 용도가 mui의 사용법을 옮기는 것보다
프로젝트에서 사용하는 컴포넌트를 확인하는 용도라고 생각해서
스토리북에 굳이 필요한 내용인지 모르겠습니다.

통일된 디자인이 아니라
체크박스의 컬러나 크기나 아이콘을 변경하면서 쓸일이 있을지 모르겠습니다.

mieumje replied:
스토리북 사용의 기본 용도를 간과한 것 같습니다 😓 죄송합니다. 
기본적인 checkbox만 남겨 두도록 수정하겠습니다.

lexie-kaia commented on src/components/Mui/MuiInput.stories.js:
```diff
@@ -162,4 +168,180 @@ export const File = () => (
+  );
+};
+
+export const StartAndEndDatePicker = () => {
```
이것도 스토리북의 목적과 관련된 이야기이긴한데
스토리북의 주된 용도는 기능 확인보다
UI 확인용이라고 생각합니다.

나중에 제이가 페이지를 구성하는데 필요한 내용이지만
스토리북에 있어야할까 생각이 듭니다.

mieumje replied:
팽님 comment에서도 확인했습니다. Date Range Picker Component를 pro 버전부터 사용 할 수 있다고 해서 Date Picker로 임의로 구성해두었는데, container로 분리하는게 맞는 것 같습니다. 수정하도록 하겠습니다.

lexie-kaia reviewed (commented):
의도를 명확하게 말하려고 하다보니
제가 남긴 리뷰가 좀 딱딱하게 들리는 것 같아 걱정이네요😢

그리고 storybook provider 관련해서 남긴
저의 pr 한번 확인 부탁드립니다.
MuiProvider를 일일이 넣지 않아도 되도록 픽스했습니다.

GeonWooPaeng commented:
제이 
공통으로 사용하는 것으로 인해 예제도 많이 넣어서 코드짜느라 고생하셨습니다. 
코드를 보면서 공부 많이 되었습니다. :) 
실행 부분에서는 크게 문제가 없었습니다.

다음에는 조금 편하게 예제를 필요한 것만 만들어도 좋다는 생각이 됩니다. :)
추가로 커스텀이 크게 필요하지 않은 Avatar 등은 stories를 만드는 것보다 그냥 사용하시는 것이 
정신 건강에 편하실 것 같습니다. :)
고생하셨습니다. !!!!

GeonWooPaeng commented on src/components/Mui/MuiInput.stories.js:
```diff
@@ -162,4 +168,180 @@ export const File = () => (
+};
+
+export const LabelCheckBoxes = () => {
+  const [checkedList, setCheckedList] = useState([]);
+  const onCheckedElement = (checked, item) => {
+    if (checked) {
+      setCheckedList([...checkedList, item]);
+    } else {
+      setCheckedList(checkedList.filter((element) => element !== item));
+    }
+  };
```
p5;
해당 부분은 container라는 생각이 듭니다. :)

mieumje replied:
mui Date range picker component가 pro 버전부터 사용가능해 Date Picker로 range picker를 만들었는데 container로 빼는게 나은 것 같습니다. 이 부분 수정하겠습니다.

mieumje committed:
Merge branch 'develop' of https://github.com/prgrms-web-devcourse/Team_KPPL_STUTI_FE into feature/공통-컴포넌트-인풋

mieumje committed:
refactor: StartAndEndDatePicker 삭제, console.log 삭제

Container로 분리하는 것이 합당해 삭제
test용 console.log 삭제

mieumje committed:
refactor: Input storybook checkbox 수정

기본적인 UI만 확인 할 수 있도록 수정

mieumje committed:
refactor: Mui Avatar component storybook 수정

MuiProvider 주입 변경 방식 수정에 따라 import 삭제

Avatar default args로 src만 수정 가능하도록 변경
argTypes를 삭제하여 Avatargroup에서 불필요한 컨트롤 주입 방지

lexie-kaia reviewed (approved):
확인!!
